### ISSUE TITLE
Show dropdown of popular tags after typing #

### ISSUE THREAD

**Issue by jaysonvirissimo** [bellroy/predictionbook#230]
Typing # in the prediction description or in a response message treats the following word as a tag, but the user in on their own when deciding which to use. Instead, we could show a dropdown upon typing #, from which they can choose an existing tag or continue to type the tag name they want if they don't see what they are looking for.

This dropdown could be drawn from:
1. The globally most used tags (that match the already typed characters, such as `state` and `stats` if they have typed `#sta`)
2. The tags they have most used themselves
3. Some combination of (1) and (2)

**Comment by Raemon** [comment]
I think "tags they have most used themselves" is probably most useful. I'm confident that using my own tags is useful, it's more speculative that other peoples' will be.

But, if it turns out to be easy, maybe prioritize my tags, and then show the top globally used tags that match whatever characters I've entered after the # symbol. (I had a guess that the UI would work fastest/most-seamlessly if it just used the tags associated with my user, which are probably a quite small number, rather to use tags associated with my user than a global search. But that's just a guess. I think having it work quickly and responsively is more important that global tag search)

**Comment by Raemon** [comment]
I guess as-written it doesn't currently specify: it'd also be nice if after typing #, the next letters I type get entered into a search process.

**Comment by jaysonvirissimo** [comment]
Twitter does something sorta like this:
<img width="600" alt="Screenshot 2023-01-14 at 10 09 25 PM" src="https://user-images.githubusercontent.com/1292454/212523970-943f5643-f1eb-499e-b299-93874323cc16.png">
It looks like they match based on site wide popularity, but cap the number in the dropdown to 8 maximum.

**Comment by Raemon** [comment]
Yeah this is a fairly common UI paradigm. I think it's not crazy to go with
either choice, but weakly predict personal-user-tags will be more
convenient.
